### PR TITLE
[LDB] xm-ms-sheduler: Implement ONE_TIME ScheduleType

### PR DESCRIPTION
**Pull request by kkyrychenko**

### CODE REVIEW

File: src/main/java/com/icthh/xm/ms/scheduler/manager/DefaultRunnableTask.java [hunk 1]
```diff
@@ -27,35 +27,87 @@ public void run() {
...
-            manager.handleTask(task);
-
-            if (afterRun != null) {
-                afterRun.accept(task);
-            }
-
-            if (isExpired()) {
-                // time to delete
-                log.info("remove expired scheduled task {} due to end date: {}", task.getId(), task.getEndDate());
+            if (isExpiredBeforeExecution()) {
```
Comment (v-kyrychenko) [inline src/main/java/com/icthh/xm/ms/scheduler/manager/DefaultRunnableTask.java:30]: simplify

if  if (isExpiredBeforeExecution()) {
....
return

File: src/main/java/com/icthh/xm/ms/scheduler/manager/DefaultRunnableTask.java [hunk 2]
```diff
@@ -27,35 +27,87 @@ public void run() {
...
                 manager.deleteExpiredTask(task);
```
Comment (v-kyrychenko) [inline src/main/java/com/icthh/xm/ms/scheduler/manager/DefaultRunnableTask.java]: remove code duplication
   manager.deleteExpiredTask(task);
                if (afterExpiry != null) {
                    afterExpiry.accept(task);
                }

File: src/main/java/com/icthh/xm/ms/scheduler/manager/SchedulingManager.java [hunk 3]
```diff
@@ -241,6 +249,10 @@ private ScheduledFuture schedule(RunnableTask expirable) {
             case CRON:
                 future = taskScheduler.schedule(expirable, new CronTrigger(task.getCronExpression()));
                 break;
+            case ONE_TIME:
+                task.setEndDate(task.getStartDate().plusSeconds(task.getTtl()));
```
Comment (v-kyrychenko) [inline src/main/java/com/icthh/xm/ms/scheduler/manager/SchedulingManager.java]: ttl is not mandatory field
please handle situation if ttl is null

**Review by v-kyrychenko — APPROVED** [review]

File: src/main/java/com/icthh/xm/ms/scheduler/manager/DefaultRunnableTask.java (hunk 2 above)
Comment (kkyrychenko) [reply src/main/java/com/icthh/xm/ms/scheduler/manager/DefaultRunnableTask.java]: +

File: src/main/java/com/icthh/xm/ms/scheduler/manager/SchedulingManager.java (hunk 3 above)
Comment (kkyrychenko) [reply src/main/java/com/icthh/xm/ms/scheduler/manager/SchedulingManager.java]: +

**Pull request merged**